### DISCUSSION TITLE
Recipes and Best practices

### DISCUSSION THREAD

**Discussion by colomain** [lihaoyi/cask#130] [Ideas]
Hi there! I've been using Cask framework for serveral months to build a saas web project... and at this point I wonder if anyone would be interested in writing a web to gather all kind of recipes and best practices on using Cask. It would be awesome to share info about Cask in a central site (github maybe),  approved by community. Decorators (in other frameworks called Middlewares), ways to serve static content, CORS, CRSF, Authentication, Logging, DB connection (besides new ScalaSql),  organizing project, serve html pages using scalatags, htmx, etc... What you think? 🤔

@lolgab also add a very interesting idea in discords channel (https://discord.com/channels/632150470000902164/940067748103487558/1250456173900070942): 

> I'm not sure what Haoyi thinks about this, but the "com-lihaoyi way" in Mill has been to have contrib modules, which are hosted in the main Mill repository, but treated a bit differently compared to the main modules, which means less backwards compatibility guarantees and lower barrier of entry for new contributions.
> I think you could contribute a PR to cask to add a new module like
> contrib.cors which would depend on the main module and provide the extra functionality.
> And then we could have more contrib modules for other features.
> It seems a great idea to me!

**Comment by lihaoyi** [comment]
@colomain I think the first step here would be to put whatever you have out there; whether code or ideas or writeups. 

Just putting them up here as github discussions would be a great start. After that, we can discuss what should go into the codebase, what should go in the docs, and what should remain external.

**Reply by colomain** [reply]
Ok, I'll do... let me some days to gather some usefull (for me) bits of code to share. Other folks could do the same here.
